SS2-Project/Assets/Starstorm2/ContentClasses/SS2Item.cs

```diff
@@ -14,14 +14,51 @@ namespace SS2
     /// <summary>
     /// <inheritdoc cref="IItemContentPiece"/>
     /// </summary>
-    public abstract class SS2Item : IItemContentPiece
+    public abstract class SS2Item : IItemContentPiece, IContentPackModifier
     {
-        public abstract NullableRef<List<GameObject>> ItemDisplayPrefabs { get; }
+        public ItemAssetCollection AssetCollection { get; private set; }
+        public NullableRef<List<GameObject>> ItemDisplayPrefabs;
+        public ItemDef ItemDef;
+
         ItemDef IContentPiece<ItemDef>.Asset => ItemDef;
-        public abstract ItemDef ItemDef { get; }
+        NullableRef<List<GameObject>> IItemContentPiece.ItemDisplayPrefabs => ItemDisplayPrefabs;
+
+        public abstract SS2AssetRequest<T> AssetRequest<T>() where T : UnityEngine.Object;
 
         public abstract void Initialize();
         public abstract bool IsAvailable(ContentPack contentPack);
-        public abstract IEnumerator LoadContentAsync();
+        public virtual IEnumerator LoadContentAsync()
+        {
+            SS2AssetRequest<UnityEngine.Object> request = AssetRequest<UnityEngine.Object>();
+
+            request.StartLoad();
+            while (!request.IsComplete)
+                yield return null;
+
+            if((ItemAssetCollection)request.Asset)
+            {
+                AssetCollection = (ItemAssetCollection)request.Asset;
+
+                ItemDef = AssetCollection.itemDef;
+                ItemDisplayPrefabs = AssetCollection.itemDisplayPrefabs;
+
+                OnAssetCollectionLoaded(AssetCollection);
+            }
+            else if((ItemDef)request.Asset)
```

Comment:
i'm around 90% sure this would throw an exception if the Asset is not of type ItemDef, I think it'd be better to use pattern matching here.

`if (reques.Asset is ItemDef itemDef)`